### ISSUE TITLE
clippy::missing-assert-message gives false positive

### ISSUE THREAD

**Issue by insideoutclub** [rust-lang/rust-clippy#13490]
### Summary

This lint shows up even when I add as message as a second parameter to the assert! macro.

### Lint Name

missing-assert-message

### Reproducer

I tried this code:

```rust
pub struct Service {
    ready: bool,
}

pub fn call(service: Service) {
    assert!(service.ready, "`service.poll_ready()` must be called first to ensure that service is ready to receive requests");
}

```

I saw this happen:

```
C:\Users\sandersd\Exercism\rust\poker\src>cargo clippy -- -Dclippy::missing_assert_message
    Checking poker v1.1.0 (C:\Users\sandersd\Exercism\rust\poker)
error: assert without any message
 --> src\lib.rs:6:5
  |
6 |     assert!(service.ready, "`service.poll_ready()` must be called first to ensure that service is ready to receive requests...
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: consider describing why the failing assert is problematic
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_assert_message
  = note: requested on the command line with `-D clippy::missing-assert-message`

error: could not compile `poker` (lib) due to 1 previous error```

I expected to see this happen:

C:\Users\sandersd\Exercism\rust\poker\src>cargo clippy -- -Dclippy::missing_assert_message
    Checking poker v1.1.0 (C:\Users\sandersd\Exercism\rust\poker)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.50s

### Version

rustc 1.81.0 (eeb90cda1 2024-09-04)
binary: rustc
commit-hash: eeb90cda1969383f56a2637cbd3037bdf598841c
commit-date: 2024-09-04
host: x86_64-pc-windows-msvc
release: 1.81.0
LLVM version: 18.1.7

### Additional Labels

_No response_

**Comment by alex-semenyuk** [comment]
Do not observe such issue at last version https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=0a03582571da3ba6c6c51682b777330c

**Comment by insideoutclub** [comment]
I can't reproduce the issue on the rust playground, but I am able to reproduce the issue from the command line on my windows machine.

**Comment by insideoutclub** [comment]
On the command line, my Cargo.toml file had the edition set to 2018 rather than 2021. With the 2021 edition, the problem is solved. Closing the case. Thank-you!

**Comment by JP-Ellis** [comment]
I believe this should be re-opened, because as @insideoutclub mentioned, this is still an issue when used against the 2018 edition.

Example reproduction: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a3ac148e38ed8651cc15f6cfb89af055

```rust
#![deny(clippy::missing_assert_message)]
#![allow(clippy::assertions_on_constants)]

fn main() {
    assert!(true, "test");
}
```

```text
    Checking playground v0.0.1 (/playground)
error: assert without any message
 --> src/main.rs:5:5
  |
5 |     assert!(true, "test");
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = help: consider describing why the failing assert is problematic
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_assert_message
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![deny(clippy::missing_assert_message)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
